### PR TITLE
GRW-1798 / Feature / Exchange authorizationCode to accessToken when connecting payments for hedvig-com

### PR DESCRIPTION
**Pull request by alebedev**
## What?

Added /new-member/exchange-token ednpoint that gets accessToken, saves it cookie, then redirects to local URL specified by `next` param

## Why?

Safe way to do cross-domain token transfer that does not rely on shared cookie and therefore works in any environment

**Ticket(s): []**

### CODE REVIEW

File: src/server/exchangeToken.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import Axios from 'axios'
+import Router from 'koa-router'
+import { ParameterizedContext } from 'koa'
+import { createSession } from 'shared/sessionStorage'
+import { ServerCookieStorage } from 'utils/storage/ServerCookieStorage'
+import { WithLoggerState } from './middleware/enhancers'
+
+enum SearchParams {
+  AuthorizationCode = 'authorizationCode',
```
Comment (robinandeer) [inline src/server/exchangeToken.ts:9]: <i class='graphite__hidden'>[Re: lines 9 to 9]<br/></i>

Didn't you define this as `authorization_code` in racoon?<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/HedvigInsurance/web-onboarding/1210?utm_source=unchanged-line-comment">Graphite</a>.</span>

File: src/server/exchangeToken.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  nextUrl.searchParams.delete(SearchParams.AuthorizationCode)
+
+  const authorizationCode = ctx.URL.searchParams.get(
+    SearchParams.AuthorizationCode,
+  )
+  if (!authorizationCode) {
+    return ctx.throw(
+      400,
+      `Search param ${SearchParams.AuthorizationCode} is required`,
+    )
```
Comment (robinandeer) [inline src/server/exchangeToken.ts:34]: What happens when a user ends up here? Just thinking it would be nice to show an error page in the future...

**Review by robinandeer — APPROVED** [review]
Looks good! Just double check the search param names so they match :)

[looksgood](https://app.graphite.dev/macros/HedvigInsurance)

File: src/server/exchangeToken.ts (hunk 1 above)
Comment (alebedev) [reply src/server/exchangeToken.ts:9]: Good catch!

**Pull request merged**